sql/flow.go

```diff
@@ -143,35 +162,43 @@ func CommonDockerUtil(cmd, args []string, flags map[string]string, mountDirs []s
 		"",
 	)
 	if err != nil {
-		return statusCode, fmt.Errorf("docker container creation failed %w", err)
+		return statusCode, "", fmt.Errorf("docker container creation failed %w", err)
 	}
 
 	if err := cli.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
-		return statusCode, fmt.Errorf("docker container start failed %w", err)
+		return statusCode, "", fmt.Errorf("docker container start failed %w", err)
 	}
 
 	statusCh, errCh := cli.ContainerWait(ctx, resp.ID, container.WaitConditionNotRunning)
 	select {
 	case err := <-errCh:
 		if err != nil {
-			return statusCode, fmt.Errorf("docker container wait failed %w", err)
+			return statusCode, "", fmt.Errorf("docker container wait failed %w", err)
 		}
 	case status := <-statusCh:
 		statusCode = status.StatusCode
 	}
 
 	cout, err := cli.ContainerLogs(ctx, resp.ID, types.ContainerLogsOptions{ShowStdout: true, ShowStderr: true})
 	if err != nil {
-		return statusCode, fmt.Errorf("docker container logs fetching failed %w", err)
+		return statusCode, "", fmt.Errorf("docker container logs fetching failed %w", err)
 	}
 
-	if _, err := Io().Copy(os.Stdout, cout); err != nil {
-		return statusCode, fmt.Errorf("docker logs forwarding failed %w", err)
+	outputLogs := ""
+	if returnStdout {
+		outputLogs, err = captureStdout(cout)
+		if err != nil {
+			return statusCode, "", err
+		}
+	} else {
+		if _, err := Io().Copy(os.Stdout, cout); err != nil {
+			return statusCode, "", fmt.Errorf("docker logs forwarding failed %w", err)
+		}
```

Comment:
Since we already copy the containers stdout to the os stdout, I would suggest we read `os.Stdout` outside of the this function.

Reply:
Returning cout instead now and dropping the usage of capturing stdout as discussed. Also, readcloser (cout) can be read only once as it acts as a stream and hence we do not copy it to stdout when we intend to return it for further usage in the calling function.

Reply:
Makes sense 👍